handlers/accesscontrol/access-control.go

```diff
@@ -379,6 +498,19 @@ func (ac *AccessControlHandlersCollection) cachePlaybackAccessControlInfo(playba
 	}
 	refreshIntervalCache.mux.Unlock()
 
+	// cache viewer limit per user data
+	viewerLimitCache.mux.Lock()
+	if gateConfig.ViewerLimitPerUser != 0 && gateConfig.UserID != "" {
+		if _, ok := viewerLimitCache.data[playbackID]; !ok {
+			viewerLimitCache.data[playbackID] = &ViewerLimitCacheEntry{}
+		}
+		viewerLimitCache.data[playbackID].ViewerLimitPerUser = gateConfig.ViewerLimitPerUser
+		viewerLimitCache.data[playbackID].UserID = gateConfig.UserID
+	} else {
+		delete(viewerLimitCache.data, playbackID)
+	}
+	viewerLimitCache.mux.Unlock()
```

Comment:
Hmm, WDYT of changing `PlaybackAccessControlEntry` to store the `GateConfig` object instead and avoiding this separate cache? Then to get the viewer limit for a given playback ID (e.g. [L378](https://github.com/livepeer/catalyst-api/pull/1296/files#diff-e4212fa5c0109280d6faf7edd53a45935b2460fca63acb0e9fadf092e1fc225bR378)), we would instead do `GetPlaybackAccessControlInfo` (which can change to return to be the full `PlaybackAccessControlEntry`, not just the boolean `Allow`). 

This would mean we change the viewer limit check to after [L328](https://github.com/livepeer/catalyst-api/pull/1296/files#diff-e4212fa5c0109280d6faf7edd53a45935b2460fca63acb0e9fadf092e1fc225bR328), which IMO makes more sense: we only check the viewer limit after fetching it (rn from what I could tell, the very first playback won't be checked against the limit right? the separate cache only gets initialized after it is used)

Reply:
Yeah, that could work this way. Though, I'm not sure I want to touch that flow right now.

This code is quite messy and very sensitive, because: 1) It makes a blocking call 2) It affects all viewers for all customers. Also, the rate limit is implemented with its own cache, so it would still be inconsistent. I'll reconsider it while removing `rate limit` code.

Reply:
Cool. Yeah, I think this only makes sense if the `rate_limit` gets refactored too. Let's keep it separate.